core/src/main/java/io/confluent/rest/RestConfig.java

```diff
@@ -33,9 +33,16 @@
   protected static final boolean DEBUG_CONFIG_DEFAULT = false;
 
   public static final String PORT_CONFIG = "port";
-  protected static final String PORT_CONFIG_DOC = "Port to listen on for new connections.";
+  protected static final String PORT_CONFIG_DOC = "DEPRECATED: port to listen on for new HTTP connections. Use " +
+      "listeners instead.";
   protected static final int PORT_CONFIG_DEFAULT = 8080;
 
+  public static final String LISTENERS_CONFIG = "listeners";
+  protected static final String LISTENERS_DOC = "List of listeners. http and https are supported. Each " +
+      "listener must include the protocol, hostname, and port. For example: http://myhost:8080," +
+      "https://0.0.0.0:8081";
+  protected static final String LISTENERS_DEFAULT = ""; // TODO: add a default value when `PORT_CONFIG` is deprecated.
```

Comment:
presumably you mean when it is removed since that is when it is safe? The alternative here is to change the default value of `port` immediately to remove the default and make `listeners` default to `http://:8080` and make any appearance of `port` force usage of the old approach. Not sure either approach is better -- only benefit I see to the alternative is that the new default is introduced immediately.